File: Python/bulder_pattern/sanduiche_builder.py
from sanduiche import Sanduiche


class SanduicheBuilder:
    def set_pao(self, pao):
        self.pao = pao
        return self
        
    def set_carne(self, carne):
        self.carne = carne
        return self
        
    def set_queijo(self, queijo):
        self.queijo = queijo
        return self
        
    def set_tem_salada(self, tem_salada):
        self.tem_salada = tem_salada
        return self
    
    def build(self):
        sanduiche = Sanduiche()
        sanduiche.pao = self.pao
        sanduiche.carne = self.carne
        sanduiche.queijo = self.queijo
        sanduiche.tem_salada = self.tem_salada
        
        return sanduiche

File: Java/com.faculdadevincit.designpatterns.bulderpattern/src/SanduicheBuilder.java
public class SanduicheBuilder {
    private String pao;
    private String carne;
    private String queijo;
    private boolean temSalada;

    public SanduicheBuilder setPao(String pao) {
        this.pao = pao;
        return this;
    }
    public SanduicheBuilder setCarne(String carne) {
        this.carne = carne;
        return this;
    }
    public SanduicheBuilder setQueijo(String queijo) {
        this.queijo = queijo;
        return this;
    }
    public SanduicheBuilder setTemSalada(boolean temSalada) {
        this.temSalada = temSalada;
        return this;
    }

    public Sanduiche Build() {
        Sanduiche sanduiche = new Sanduiche();
        sanduiche.setPao(pao);
        sanduiche.setCarne(carne);
        sanduiche.setQueijo(queijo);
        sanduiche.setTemSalada(temSalada);

        return sanduiche;
    }
}
